src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -489,8 +490,51 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
-    public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
-        return this.redeemScriptParser.getSigInsertionIndex(hash, signingKey);
+    public int getSigInsertionIndex(Sha256Hash hashForSignature, BtcECKey signingKey) {
+        // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
+        // and any placeholder OP_0 sigs.
+
+
+        // In order to affect consensus, we keep returning zero as the default value for the insertion index
+        // when the script is not a scriptSig.
+        final int defaultSigInsertionIndexForNoScriptSig = 0;
+        String noScriptSigFormatExceptionMessage = "Script does not match ScriptSig format";
+        if (chunks.size() < 2) {
+            log.debug(noScriptSigFormatExceptionMessage);
+            return defaultSigInsertionIndexForNoScriptSig;
+        }
+
+        List<ScriptChunk> chunksWithoutRedeemScript = chunks.subList(1, chunks.size() - 1);
+        ScriptChunk redeemScriptChunk = chunks.get(chunks.size() - 1);
+
+        if (redeemScriptChunk.data == null) {
+            log.debug(noScriptSigFormatExceptionMessage);
+            return defaultSigInsertionIndexForNoScriptSig;
+        }
+
+        Script redeemScript = new Script(redeemScriptChunk.data);
+        RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(redeemScript.getChunks());
+
+        if (!redeemScriptParser.getScriptType().equals(RedeemScriptParser.ScriptType.REDEEM_SCRIPT)){
+            log.debug("Redeem script could not be parsed");
+            return defaultSigInsertionIndexForNoScriptSig;
+        }
+
+        int potentialSigInsertionIndex = 0;
```

Comment:
why `potential`? i would just call it `sigInsertionIndex`

Reply:
Updated